pkg/query/querier.go

```diff
@@ -208,6 +211,47 @@ type seriesServer struct {
 	seriesSet      []storepb.Series
 	seriesSetStats storepb.SeriesStatsCounter
 	warnings       []string
+	symbolTables   []map[uint64]string
+
+	compressedSeriesSet []storepb.CompressedSeries
+}
+
+func (s *seriesServer) DecompressSeries() error {
+	for _, cs := range s.compressedSeriesSet {
+		newSeries := &storepb.Series{
+			Chunks: cs.Chunks,
+		}
+
+		lbls := labels.Labels{}
+
+		for _, cLabel := range cs.Labels {
+			var name, val string
+			for _, symTable := range s.symbolTables {
+				if foundName, ok := symTable[uint64(cLabel.NameRef)]; ok {
+					name = foundName
+				}
+
+				if foundValue, ok := symTable[uint64(cLabel.ValueRef)]; ok {
+					val = foundValue
+				}
```

Comment:
Do we need to break the look if we found both name and value?

Reply:
:+1: good point, will add